aptos-move/aptos-vm/src/verifier/uses_randomness.rs

```diff
@@ -0,0 +1,33 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::move_vm_ext::SessionExt;
+use aptos_types::transaction::EntryFunction;
+use move_binary_format::{
+    errors::{Location, VMResult},
+    CompiledModule,
+};
+
+/// Returns true if function has an annotation that it uses randomness.
+pub(crate) fn does_entry_function_use_randomness(
+    session: &mut SessionExt,
+    entry_fn: &EntryFunction,
+) -> VMResult<bool> {
+    let module_bytes = session.load_module(entry_fn.module())?;
+    let module = CompiledModule::deserialize_with_config(
+        &module_bytes,
+        &session.get_vm_config().deserializer_config,
+    )
+    .map_err(|e| e.finish(Location::Undefined))?;
```

Comment:
Wonder if this would have perf issues

Reply:
Yeah, single node perf is failing
https://github.com/aptos-labs/aptos-core/actions/runs/7993394538/job/21829130781?pr=12134